### PR TITLE
STS enabled by default

### PR DESCRIPTION
**Pull request by pjuarezd**

### CODE REVIEW

File: resources/base/deployment.yaml [hunk 1]
```diff
@@ -40,7 +40,7 @@ spec:
             - name: MINIO_CONSOLE_TLS_ENABLE
               value: "off"
             - name: OPERATOR_STS_ENABLED
-              value: "off"
+              value: "on"
```
Comment (harshavardhana) [inline resources/base/deployment.yaml:43]: what is the impact?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: resources/base/deployment.yaml (hunk 1 above)
Comment (pjuarezd) [reply resources/base/deployment.yaml:43]: STS webserver will start when operator containers start

Comment (harshavardhana) [reply resources/base/deployment.yaml:43]: Okay so by default this service will be running, and it's public facing ?

Comment (pjuarezd) [reply resources/base/deployment.yaml:43]: Is kubernetes cluster internal, service type is ClusterIP, will not be exposed oustide the cluster unless explictly modified the service.

https://github.com/minio/operator/blob/6c3d60899715c7c19d15773e8310d97e17508e11/resources/base/service.yaml#L19-L33

The webserver is the STS endpoint where k8s apps (pods) will ask for IAM credenials.
https://github.com/minio/operator/blob/master/docs/STS.md

Policy-serviceaccount relation to provide sts credentials is managed by the [PolicyBinding](https://github.com/minio/operator/blob/master/docs/policybinding_crd.adoc) resource.

All that the `OPERATOR_STS_ENABLED =on` flag does is enable the webserver that handle the sts calls

**Review by harshavardhana — APPROVED** [review]

File: resources/base/deployment.yaml (hunk 1 above)
Comment (jiuker) [inline resources/base/deployment.yaml:43]: Shall we switch it on helm charts?

**Review by jiuker — CHANGES_REQUESTED** [review]

**Review by jiuker — DISMISSED** [review]

File: resources/base/deployment.yaml (hunk 1 above)
Comment (pjuarezd) [reply resources/base/deployment.yaml:43]: Totally, added to values.yaml

**Review by jiuker — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
Looks like this doc must be updated https://github.com/minio/operator/blob/master/docs/STS.md with newer behavior.

**Comment by pjuarezd** [comment]
> It looks like this doc must be updated https://github.com/minio/operator/blob/master/docs/STS.md with newer behavior. This is not an alpha state anymore.

Sure!, I am about to open anoter PR that will also change the api version from `sts.min.io/v1alpha1` to `sts.min.io/v1beta1`

**Pull request merged**